### ISSUE TITLE
Rolling upgrade, longer client downtime (50 s or more)

### ISSUE THREAD

**Issue by Samusername** [zalando/postgres-operator#840]
Mode: Leader-**sync**-async
synchronous_mode and synchronous_mode_strict were enabled.

Downtime, from client point of view, from around: 
16:15:41
until:
16:16:38
--> Downtime: around 55 s, etc.

In rolling upgrade, sequence:

First: acid-test-cluster-1 gets restarted and starts up. 
Second: acid-test-cluster-2 gets restarted and starts up.
Third: acid-test-cluster-0 gets restarted and starts up.

DB nodes are e.g. in following states for a while when longer downtime happened.
One Leader, but neither of the other db nodes are in "Sync Standby" role. --> Causes the downtime.

| acid-test-cluster | acid-test-cluster-0 |  192.168.0.105  |        | **stopped** |    |   unknown |  <--- **Not in "Sync Standby"**
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 | Leader | running |  8 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |        | running |    |   unknown |    <--- **Not in "Sync Standby"**
Then next, quickly changes to: 
Fri Feb 21 14:15:59 UTC 2020:

| acid-test-cluster | acid-test-cluster-0 |       None      |        |         |    |   unknown |   <--- **Not in "Sync Standby"**
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 | Leader | running |  8 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |        | **stopped** |    |   **unknown** |   <--- **Not in "Sync Standby"**

acid-test-cluster-2 stays in stopped,unknown state for a very long time. (Not good.) 
Correct behavior would be: 
acid-test-cluster-2 should be in "Sync Standby" state while acid-test-cluster-0 gets restarted, and when acid-test-cluster-1 is leader.
Problem: But -2 is in stopped,unknown state instead.

Could such situation be tried to be avoided.

Longer logs.
Restart sequence: pod "-1", pod "-2", pod "-0", 
during rolling upgrade, kubectl get pods -w:

```
Fri Feb 21 14:13:23 UTC 2020
NAME                                                     READY   STATUS    RESTARTS   AGE
acid-test-cluster-0                                      1/1     Running   0          96m
acid-test-cluster-1                                      1/1     Running   0          94m
acid-test-cluster-2                                      1/1     Running   0          95m
pg-operator-patroni-postgres-operator-6c897df876-7zns5   1/1     Running   0          28h
acid-test-cluster-2                                      1/1     Running   0          95m
acid-test-cluster-1                                      1/1     Running   0          94m
acid-test-cluster-0                                      1/1     Running   0          96m
acid-test-cluster-1                                      1/1     Terminating   0          94m
acid-test-cluster-0                                      1/1     Running       0          96m
acid-test-cluster-1                                      1/1     Terminating   0          94m
acid-test-cluster-1                                      1/1     Terminating   0          94m
acid-test-cluster-1                                      1/1     Terminating   0          94m
acid-test-cluster-1                                      0/1     Terminating   0          94m
acid-test-cluster-1                                      0/1     Terminating   0          94m
acid-test-cluster-2                                      1/1     Running       0          95m
acid-test-cluster-0                                      1/1     Running       0          96m
acid-test-cluster-1                                      0/1     Terminating   0          95m
acid-test-cluster-1                                      0/1     Terminating   0          95m
acid-test-cluster-1                                      0/1     Pending       0          0s
acid-test-cluster-1                                      0/1     Pending       0          0s
acid-test-cluster-1                                      0/1     Init:0/1      0          0s
acid-test-cluster-2                                      1/1     Running       0          95m
acid-test-cluster-0                                      1/1     Running       0          96m
acid-test-cluster-2                                      1/1     Running       0          96m
acid-test-cluster-0                                      1/1     Running       0          97m
acid-test-cluster-2                                      1/1     Running       0          96m
acid-test-cluster-1                                      0/1     PodInitializing   0          24s
acid-test-cluster-1                                      1/1     Running           0          25s
acid-test-cluster-0                                      1/1     Running           0          97m
acid-test-cluster-2                                      1/1     Running           0          96m
acid-test-cluster-0                                      1/1     Running           0          97m
acid-test-cluster-2                                      1/1     Running           0          96m
acid-test-cluster-1                                      1/1     Running           0          43s
acid-test-cluster-1                                      1/1     Running           0          47s
acid-test-cluster-2                                      1/1     Terminating       0          96m
acid-test-cluster-2                                      1/1     Terminating       0          96m
acid-test-cluster-2                                      1/1     Terminating       0          96m
acid-test-cluster-0                                      1/1     Running           0          97m
acid-test-cluster-2                                      1/1     Terminating       0          96m
acid-test-cluster-2                                      0/1     Terminating       0          96m
acid-test-cluster-2                                      0/1     Terminating       0          96m
acid-test-cluster-1                                      1/1     Running           0          57s
acid-test-cluster-2                                      0/1     Terminating       0          96m
acid-test-cluster-2                                      0/1     Terminating       0          96m
acid-test-cluster-2                                      0/1     Pending           0          0s
acid-test-cluster-2                                      0/1     Pending           0          1s
acid-test-cluster-2                                      0/1     Init:0/1          0          1s
acid-test-cluster-0                                      1/1     Running           0          97m
acid-test-cluster-1                                      1/1     Running           0          67s
acid-test-cluster-0                                      1/1     Running           0          97m
acid-test-cluster-2                                      0/1     PodInitializing   0          12s
acid-test-cluster-2                                      1/1     Running           0          13s
acid-test-cluster-1                                      1/1     Running           0          77s
acid-test-cluster-0                                      1/1     Running           0          98m
acid-test-cluster-1                                      1/1     Running           0          87s
acid-test-cluster-2                                      1/1     Running           0          31s
acid-test-cluster-0                                      1/1     Running           0          98m
acid-test-cluster-1                                      1/1     Running           0          97s
acid-test-cluster-2                                      1/1     Running           0          39s
acid-test-cluster-0                                      1/1     Running           0          98m
acid-test-cluster-0                                      1/1     Running           0          98m
acid-test-cluster-1                                      1/1     Running           0          102s
acid-test-cluster-1                                      1/1     Running           0          102s
acid-test-cluster-0                                      1/1     Terminating       0          98m
acid-test-cluster-0                                      1/1     Terminating       0          98m
acid-test-cluster-0                                      1/1     Terminating       0          98m
acid-test-cluster-0                                      1/1     Terminating       0          98m
acid-test-cluster-1                                      1/1     Running           0          103s
acid-test-cluster-2                                      1/1     Running           0          45s
acid-test-cluster-0                                      0/1     Terminating       0          98m
acid-test-cluster-0                                      0/1     Terminating       0          98m
acid-test-cluster-0                                      0/1     Terminating       0          98m
acid-test-cluster-0                                      0/1     Terminating       0          98m
acid-test-cluster-0                                      0/1     Pending           0          0s
acid-test-cluster-0                                      0/1     Pending           0          0s
acid-test-cluster-0                                      0/1     Init:0/1          0          0s
acid-test-cluster-1                                      1/1     Running           0          113s
acid-test-cluster-2                                      1/1     Running           0          55s
acid-test-cluster-1                                      1/1     Running           0          2m3s
acid-test-cluster-0                                      0/1     PodInitializing   0          23s
acid-test-cluster-0                                      1/1     Running           0          25s
```

States, sequence, 
some points when some changes happen, are in following monitored logs.

Please note especially following long lasting state:
acid-test-cluster-2 |  ...   | stopped |    |   unknown |  <--- "stopped", "unknown"

Longer logs, during the whole rolling upgrade.
There are some comments also on the right. (May be hidden in browser, needs scrolling to right.)

```
Fri Feb 21 14:13:37 UTC 2020
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |     Role     |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.105  |    Leader    | running |  7 |           |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.221 |              | running |  7 |         0 |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.37 | Sync Standby | running |  7 |         0 |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
...
Later:

+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |     Role     |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.105  |    Leader    | running |  7 |           |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.221 |              | running |  7 |         0 |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.37 | Sync Standby | running |  7 |         0 |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+

acid-test-cluster-1 is restated (good):

Fri Feb 21 14:13:54 UTC 2020
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |     Role     |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.105  |    Leader    | running |  7 |           |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.221 |              | stopped |    |   unknown |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.37 | Sync Standby | running |  7 |        33 |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+

...
Later:

+-------------------+---------------------+----------------+--------------+---------+----+-----------+
|      Cluster      |        Member       |      Host      |     Role     |  State  | TL | Lag in MB |
+-------------------+---------------------+----------------+--------------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 | 192.168.0.105  |    Leader    | running |  7 |           |
| acid-test-cluster | acid-test-cluster-1 |      None      |              |         |    |   unknown |
| acid-test-cluster | acid-test-cluster-2 | 192.168.212.37 | Sync Standby | running |  7 |        37 |
+-------------------+---------------------+----------------+--------------+---------+----+-----------+
Fri Feb 21 14:14:06 UTC 2020

...

Later:
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |     Role     |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.105  |    Leader    | running |  7 |           |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 |              | stopped |    |   unknown |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.37 | Sync Standby | running |  7 |         0 |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
Fri Feb 21 14:14:48 UTC 2020

...

+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |     Role     |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.105  |    Leader    | running |  7 |           |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 |              | running |  7 |       279 |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.37 | Sync Standby | running |  7 |         0 |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
Fri Feb 21 14:14:54 UTC 2020
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |  Role  |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.105  | Leader | running |  7 |           |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 |        | running |  7 |       344 |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.37 |        | running |  7 |        47 |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
...
Fri Feb 21 14:14:57 UTC 2020
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |  Role  |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.105  | Leader | running |  7 |           |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 |        | running |  7 |       344 |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.37 |        | stopped |    |   unknown |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
...
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |  Role  |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.105  | Leader | running |  7 |           |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 |        | running |  7 |         0 |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.37 |        | stopped |    |   unknown |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
error: unable to upgrade connection: container not found ("postgres")
Fri Feb 21 14:15:04 UTC 2020
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |     Role     |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.105  |    Leader    | running |  7 |           |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 | Sync Standby | running |  7 |        18 |
| acid-test-cluster | acid-test-cluster-2 |       None      |              |         |    |   unknown |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
...
Fri Feb 21 14:15:11 UTC 2020
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |     Role     |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.105  |    Leader    | running |  7 |           |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 | Sync Standby | running |  7 |         0 |
| acid-test-cluster | acid-test-cluster-2 |       None      |              |         |    |   unknown |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
...
Fri Feb 21 14:15:23 UTC 2020
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |     Role     |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.105  |    Leader    | running |  7 |           |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 | Sync Standby | running |  7 |        21 |
| acid-test-cluster | acid-test-cluster-2 |       None      |              |         |    |   unknown |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
...
Fri Feb 21 14:15:37 UTC 2020
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |     Role     |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.105  |    Leader    | running |  7 |           |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 | Sync Standby | running |  7 |        20 |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |              | stopped |    |   unknown |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
...

NOTE: downtime starts around here. Client: "was aborted: FATAL: terminating connection due to administrator command"

Fri Feb 21 14:15:43 UTC 2020
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |     Role     |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.105  |    Leader    | running |    |           |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 | Sync Standby | running |  7 |         9 |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |              | running |  7 |       230 |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
...
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |     Role     |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.105  |    Leader    | running |    |           |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 | Sync Standby | running |  7 |         9 |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |              | running |  7 |       230 |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
Fri Feb 21 14:15:46 UTC 2020
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |  Role  |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.105  |        | stopped |    |   unknown |     <----- stopped 
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 | Leader | running |  7 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |        | running |  7 |       230 |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
...
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |  Role  |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.105  |        | stopped |    |   unknown |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 | Leader | running |  8 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |        | running |  7 |       256 |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
Fri Feb 21 14:15:49 UTC 2020
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |  Role  |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.105  |        | stopped |    |   unknown |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 | Leader | running |  8 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |        | running |    |   unknown |    <--- UNKNOWN Lag in MB
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
...
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |  Role  |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.105  |        | stopped |    |   unknown |    
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 | Leader | running |  8 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |        | running |    |   unknown |    <--- UNKNOWN Lag in MB
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
Fri Feb 21 14:15:54 UTC 2020
...
Fri Feb 21 14:15:59 UTC 2020
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |  Role  |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |       None      |        |         |    |   unknown |   <--- empty state
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 | Leader | running |  8 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |        | stopped |    |   unknown |   <--- stopped
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
...
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |  Role  |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |       None      |        |         |    |   unknown |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 | Leader | running |  8 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |        | stopped |    |   unknown |  <--- "stopped", "unknown" all the time
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
Fri Feb 21 14:16:06 UTC 2020
...
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |  Role  |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |       None      |        |         |    |   unknown |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 | Leader | running |  8 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |        | stopped |    |   unknown |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
Fri Feb 21 14:16:32 UTC 2020
...
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |  Role  |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |       None      |        |         |    |   unknown |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 | Leader | running |  8 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |        | stopped |    |   unknown |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
Fri Feb 21 14:16:40 UTC 2020
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |  Role  |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.106  |        | stopped |    |   unknown |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 | Leader | running |  8 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |        | stopped |    |   unknown |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
...
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |  Role  |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.106  |        | running |  8 |         0 |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 | Leader | running |  8 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |        | stopped |    |   unknown |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
Fri Feb 21 14:16:43 UTC 2020
...
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |  Role  |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.106  |        | running |  8 |         0 |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 | Leader | running |  8 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |        | stopped |    |   unknown |
+-------------------+---------------------+-----------------+--------+---------+----+-----------+
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |     Role     |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.106  | Sync Standby | running |  8 |        26 |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 |    Leader    | running |  8 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |              | stopped |    |   unknown |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
Fri Feb 21 14:16:48 UTC 2020
...
Fri Feb 21 14:16:51 UTC 2020
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |     Role     |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.106  | Sync Standby | running |  8 |         0 |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 |    Leader    | running |  8 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |              | stopped |    |   unknown |   <--- Still in "stopped", "unknown" state.
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
...

acid-test-cluster-2 still continues to be in "stopped" or "stopped"&"unknown" state, all the time (long time).

Fri Feb 21 14:17:14 UTC 2020
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |     Role     |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.106  | Sync Standby | running |  8 |         0 |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 |    Leader    | running |  8 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |              | stopped |    |   unknown |  <--- Still in "stopped", "unknown" state.
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+


Fri Feb 21 14:17:20 UTC 2020
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |     Role     |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.106  | Sync Standby | running |  8 |        40 |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 |    Leader    | running |  8 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |              | stopped |    |   unknown |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |     Role     |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.106  | Sync Standby | running |  8 |         0 |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 |    Leader    | running |  8 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |              | stopped |    |   unknown |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
...

Fri Feb 21 14:18:28 UTC 2020
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |     Role     |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.106  | Sync Standby | running |  8 |        48 |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 |    Leader    | running |  8 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |              | stopped |    |   unknown |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+

...
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |     Role     |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.106  | Sync Standby | running |  8 |        29 |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 |    Leader    | running |  8 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |              | stopped |    |   unknown |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |     Role     |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.106  | Sync Standby | running |  8 |        29 |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 |    Leader    | running |  8 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |              | running |  7 |       768 |  <--- Finally "running"
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
Fri Feb 21 14:18:40 UTC 2020

...

Fri Feb 21 14:18:48 UTC 2020
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |     Role     |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.106  | Sync Standby | running |  8 |        29 |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 |    Leader    | running |  8 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |              | running |  8 |         0 |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
...
Fri Feb 21 14:18:51 UTC 2020
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
|      Cluster      |        Member       |       Host      |     Role     |  State  | TL | Lag in MB |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+
| acid-test-cluster | acid-test-cluster-0 |  192.168.0.106  | Sync Standby | running |  8 |         0 |
| acid-test-cluster | acid-test-cluster-1 | 192.168.209.222 |    Leader    | running |  8 |           |
| acid-test-cluster | acid-test-cluster-2 |  192.168.212.38 |              | running |  8 |         0 |
+-------------------+---------------------+-----------------+--------------+---------+----+-----------+

```

**Comment by FxKu** [comment]
Thanks for providing such detailed test results. It's a bit overwhelming with all your tickets I have to admit, but we can have a closer look. Are you maybe interested in contributing to our [e2e tests](https://github.com/zalando/postgres-operator/tree/master/e2e)?

Just by quickly scrolling through your printed tables I see nothing here to really worry about. You are talking about client downtime connected only to one instance? Because database downtime with no master is usually only around 3-5 seconds. Your client needs to reconnect then or you're reading from the replica then.

**Comment by CyberDem0n** [comment]
Please also keep in mind that patronictl list output doesn't prove anything because it is showing a bit stale information, up to 10 seconds.

In your synchronous setup transactions will not be executed until at least one replica will connect and become synchronous. This is also not an instant process because after the failover every replica must be restarted and it takes time.

Last, but not least, the smaller downtime you want to have - the more it will cost you. Think about it. So far you didn't invested a penny but have requirements for a million.

**Comment by Samusername** [comment]
Yes, I was thinking someone may wonder about the (amount) of tickets and questions.
But I / we have not had yet time for implementations/contribution otherwise.
I am first trying to check what looks like a potential alternative, and which could possibly have wider audience in the future, or which would at least be good (enough). SQL DB system in k8s, with HA, etc. (Even auto healing, etc. And more! ) Postgres would be preferred.
It is hard to find such (good one), even from fully commercial alternatives (and not only Postgres), which would fulfil all the needs. And there are many alternatives! Stability is one topic also. My current time has been spent on such topics. (Sorry for that.) For example performance tuning was essential to even proceed. The work amount with such topics is "overwhelming".

While I was checking functionalities from your alternative, I decided to already send information to the "forum"/"GitHub", when I noticed something worthwhile. I thought, maybe it is worthwhile to already mention such topics, when I noticed such. Some are testing results, where the testing itself requires time also. Maybe some other expert might see those worth implementation, and perhaps have will to try to implement something.

It may be possible that I / someone could contribute more, to functionalities, after "comparisons"/checking features.
